GVFS/GVFS.Common/Git/GitSsl.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text.RegularExpressions;
+using GVFS.Common.Tracing;
+
+namespace GVFS.Common.Git
+{
+    public class GitSsl
+    {
+        public readonly string CertificatePathOrSubjectCommonName;
+        public readonly bool IsCertificatePasswordProtected;
+        public readonly bool ShouldVerify;
+
+        public GitSsl()
+        {
+            this.CertificatePathOrSubjectCommonName = null;
+            this.IsCertificatePasswordProtected = false;
+            this.ShouldVerify = true;
+        }
+
+        public GitSsl(IDictionary<string, GitConfigSetting> configSettings) : this()
+        {
+            if (configSettings != null)
+            {
+                if (configSettings.TryGetValue(GitConfigSetting.HttpSslCert, out GitConfigSetting sslCerts))
+                {
+                    this.CertificatePathOrSubjectCommonName = sslCerts.Values.Single();
+                }
+
+                if (configSettings.TryGetValue(GitConfigSetting.HttpSslCertPasswordProtected, out GitConfigSetting isSslCertPasswordProtected))
+                {
+                    this.IsCertificatePasswordProtected = isSslCertPasswordProtected.Values.Select(bool.Parse).Single();
+                }
+
+                if (configSettings.TryGetValue(GitConfigSetting.HttpSslVerify, out GitConfigSetting sslVerify))
+                {
+                    this.ShouldVerify = sslVerify.Values.Select(bool.Parse).Single();
+                }
+            }
+        }
+
+        public string GetCertificatePassword(ITracer tracer, GitProcess git)
+        {
+            if (git.TryGetCertificatePassword(tracer, this.CertificatePathOrSubjectCommonName, out string password, out string error))
+            {
+                return password;
+            }
+
+            return null;
+        }
+
+        public X509Certificate2 GetCertificate(ITracer tracer, string certificatePassword, bool onlyLoadValidCertificateFromStore)
+        {
+            EventMetadata metadata = new EventMetadata
+            {
+                { "CertificatePathOrSubjectCommonName", this.CertificatePathOrSubjectCommonName },
+                { "isPasswordSpecified", string.IsNullOrEmpty(certificatePassword) },
+                { "shouldVerify", onlyLoadValidCertificateFromStore }
+            };
+
+            var result =
+                this.GetCertificateFromFile(tracer, metadata, certificatePassword, onlyLoadValidCertificateFromStore) ??
+                this.GetCertificateFromStore(tracer, metadata, onlyLoadValidCertificateFromStore);

```

Comment:
We should be able to remove the `onlyLoadValidCertificateFromStore` variable and use the `ShouldVerify` property right?